core/src/main/scala/org/apache/spark/util/NextIterator.scala

```diff
@@ -60,8 +60,10 @@ private[spark] abstract class NextIterator[U] extends Iterator[U] {
    */
   def closeIfNeeded() {
     if (!closed) {
```

Comment:
By the way, this use of `nextIterator()` explains why this bug was able to occur on 1.4.x: the first close() call, which throws an exception, comes from `NextIterator`, and the second comes from the task completion listener.